### PR TITLE
initial docker support

### PR DESCRIPTION
**Pull request by estenberg**
Not sure about the right place to put this, I don't see Dockerfile in other repos (e.g. integration).
@kacf maybe you know if this is appropriate?

### CODE REVIEW

**Comment by kacf** [comment]
`integration` doesn't have any Dockerfile because it is only a user of already existing Docker containers. It's more appropriate to looks inside the services themselves, like [this one](https://github.com/mendersoftware/useradm/blob/master/Dockerfile). So yeah, at least to me this is appropriate. I'll let the mender-convert guys review the details though.

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Create a directory `input` under the directory where you copied these files (`build.sh`, `run.sh`, etc.):
+
+```bash
+mkdir input
+```
+
+Put your raw disk image into `input`, e.g. `input/2018-10-09-raspbian-stretch.img`.
+Then put a Mender client binary compiled for your device into `input/mender-1.6.0-arm-linux-gnueabihf-gcc`.
+See [the Mender documentation on how to cross-compile the Mender client for your device](https://docs.mender.io/development/client-configuration/cross-compiling).
```
Comment (mirzak) [inline docker/README.md]: Would be nice to add this to the container as well, either pre-built "mender" binary or the steps necessary to compile it.

Might be an optimization for later.

Comment (DominikAdamski) [reply docker/README.md]: @estenberg  In my opinion we should firstly perform manual conversion inside docker, and then try to add scripts to do it in automatic way. Adding initial support for docker without checking everything carefully can cost us a lot of effort without any significant results.

Comment (estenberg) [reply docker/README.md]: @mirzak agreed, we need to make the mender client packaging/compilation easier. This was just the first step in many. :) I was also planning to
* add support for taking everything as arguments to the container (either ARG or ENV), so we don't have to work inside the container itself
* publish it on docker hub somewhere, so users don't have to build it
* get the right Mender client for the board, like you mention. This is the most "hairy" one, as it needs to match the architecture (at least ARM vs. x86). Any help here appreciated! :)

**Comment by estenberg** [comment]
@kacf thanks! looking at the useradm reference, maybe this should even be moved to the root directory (one level up)?

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    bison \
+    flex \
+    mtools \
+    parted \
+    mtd-utils \
+    e2fsprogs \
+    u-boot-tools \
+    pigz \
+    device-tree-compiler \
+    autoconf \
```
Comment (apodogrocki) [inline docker/Dockerfile]: add:
autotools-dev \
libtool \
pkg-config \

**Comment by estenberg** [comment]
Thanks for the comments everyone. Dependencies updated and verified it builds for both BBB and Rpi3. However, testing it on real hw is still pending, so should be considered experimental.

**Pull request merged**